syntax = "proto3";

package consensus;

option go_package = "./pb";

service RaftNode {
  rpc Propose(Command) returns (ProposeResponse);
}

service StateMachine {
  rpc Apply(Command) returns (ApplyResponse);
}

message Command {
  string op = 1;    // "SET", "DELETE"
  string key = 2;
  string value = 3;
  bytes data = 4;   // Serialization wrapper
}

message ProposeResponse {
  bool success = 1;
  string error = 2;
}

message ApplyResponse {
  bool success = 1;
}